servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -1193,6 +1229,30 @@ void MeshStorage::_mesh_surface_generate_version_for_input_mask(Mesh::Surface::V
 
 		attributes.push_back(vd);
 		buffers.push_back(buffer);
+
+		// Since the previous vertex, normal and tangent can't be part of the vertex format but they are required when motion
+		// vectors are enabled, we opt to push a copy of the vertex attribute with a different location and buffer (if it's
+		// part of an instance that has one).
+		switch (i) {
+			case RS::ARRAY_VERTEX: {
+				vd.location = PREVIOUS_LOCATION_VERTEX;
+			} break;
+			case RS::ARRAY_NORMAL: {
+				vd.location = PREVIOUS_LOCATION_NORMAL;
+			} break;
+			case RS::ARRAY_TANGENT: {
+				vd.location = PREVIOUS_LOCATION_TANGENT;
+			} break;
+		}
+
+		if (int(vd.location) != i) {
+			if (mis) {
+				buffer = mis->vertex_buffer[mis->previous_buffer];
+			}
+
+			attributes.push_back(vd);
+			buffers.push_back(buffer);
+		}
```

Comment:
This smells a bit hacky but it's the best way I could find to get around the lack of being able to modify the vertex array enumeration itself. Juan recommended against that due to breaking compatibility.

This approach also does not require any changes to the current RD design.

Reply:
This code should only be run when using skeletons. Right now it will always add the extra attributes, which we don't want

Reply:
I'll have to verify again, but I think it was necessary to add this regardless of skeletons being used because we don't have a shader permutation for meshes that only use skeletons (otherwise the validation will complain the input is not bound for regular terrain).

We should double check that.

Reply:
We have a system to set default values in case you use a mesh without a particular attribute but the shader reads the attribute. For example, you could have a mesh without vertex colors, but your shader can still read from the color attribute.

Reply:
After review we determined it is necessary to push the attributes regardless of whether it has an skeleton or not, but it's been addressed the attributes won't be added unless motion vectors are required by the renderer.